src/Stream/SourceFilter.h

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This code has been produced by the European Spallation Source
+// and its partner institutes under the BSD 2 Clause License.
+//
+// See LICENSE.md at the top level for license information.
+//
+// Screaming Udder!                              https://esss.se
+
+#pragma once
+
+#include "FlatbufferMessage.h"
+#include "Metrics/Metric.h"
+#include "Metrics/Registrar.h"
+#include "Stream/MessageWriter.h"
+#include <chrono>
+
+namespace Stream {
+
+using time_point = std::chrono::system_clock::time_point;
+uint64_t toNanoSec(time_point Time);
+
+class SourceFilter {
+public:
+  SourceFilter() = default;
+  SourceFilter(time_point StartTime, time_point StopTime,
+               MessageWriter *Destination, Metrics::Registrar RegisterMetric);
+  virtual ~SourceFilter();
+  void addDestinationPtr(Message::DestPtrType NewDestination) {
+    DestIDs.push_back(NewDestination);
+  };
+
+  virtual bool filterMessage(FileWriter::FlatbufferMessage &&InMsg);
+  void setStopTime(time_point StopTime);
+  time_point getStopTime() const { return Stop; }
```

Comment:
I am aware. The function is used by code that will be added through an upcoming PR.

Reply:
I think you're pulling these PRs out of another branch which maybe contains the code that uses it? If that is the case I guess the risk of this remaining dead code is low. I think we should try to avoid temporary dead code like this where possible though, particularly when cppcheck isn't providing a safety net for it, for when we merge v3 back to master.